test/e2e/lib/gutenberg/gutenberg-editor-sidebar-component.js

```diff
@@ -245,25 +245,15 @@ export default class GutenbergEditorSidebarComponent extends AsyncBaseContainer
 		const trashLocator = By.css( 'button.editor-post-trash' );
 
 		await this.selectDocumentTab();
-		await driverHelper.waitUntilElementLocatedAndVisible( this.driver, trashLocator );
 		await driverHelper.clickWhenClickable( this.driver, trashLocator );
 
-		// wait for 'Move to trash' button to disappear
-		try {
-			return await driverHelper.waitUntilElementNotLocated(
-				this.driver,
-				trashLocator,
-				this.explicitWaitMS * 3
-			);
-		} catch ( e ) {
-			// if it's still present send slack notification, post is deleted
-			return await SlackNotifier.warn(
-				'"Move to trash" button is still present, but post is deleted.',
-				{
-					suppressDuplicateMessages: true,
-				}
-			);
-		}
+		/**
+		 * After trashing the post, user is immediately redirected to the posts page
+		 * in Calypso. WebDriver isn't aware that the context has changed so we need
+		 * to do it manually.
+		 */
+		await this.driver.switchTo().defaultContent();
```

Comment:
I wonder if we could wrap this into a `switchToCalypsoContext` helper method for the sake of clarity. I often have to check the docs to remember what this method does exactly :)

Reply:
Not really, it just switches to the topmost frame - not necessarily Calypso I guess!